### PR TITLE
[Machine-learning][ACR-1382] document for machine-learning-inference

### PR DESCRIPTION
**Pull request by helloahn**
it's related to

- Native Machine-Learning API
- ACR: ACR-1382, ACR-1456

Signed-off-by: HyoungJoo Ahn hello.ahn@samsung.com

### CODE REVIEW

**Comment by mijin85cho** [comment]
This PR is just branch-changed version of #892, so I assigned @divyk, the reviewer of #892.

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+  ```
+
+  After configuring it, release the handle is required.
+
+  ```bash
+    status = ml_pipeline_src_release_handle (srchandle);
+  ```
+
+* #### Sink <a name="sinkelement"></a>
+  In case of getting the output tensor raw data, the configuration of data sink element is required.
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase to:
The configuration of the data sink element is required to get the output tensor raw data:

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+  ```bash
+    gchar * pipeline = g_strdup ("videotestsrc num-buffers=3 ! videoconvert ! tensor_converter ! appsink name=sinkx");
+  ```
+
+  Through `ml_pipeline_sink_register ()`, `appsink` element with the name `sinkx` become reachable.
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase to:

`appsink` element with the name `sinkx` becomes reachable through `ml_pipeline_sink_register ()`:

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+  ```bash
+    int status;
+    ml_pipeline_h handle;
+    ml_pipeline_sink_h sinkhandle;
+
+    status = ml_pipeline_sink_register (handle, "sinkx", sink_callback, user_data, &sinkhandle);
+  ```
+
+  `sink_callback ()` is called whenever appsink, `sinkx`, receives data. with [`data`](#tensorsdata), [`info`](#tensorsinfo), we can get the output data easily, and `user_data` can be used in the callback.
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase to:

You can call `sink_callback ()`, whenever `appsink` and `sinkx` receives data. You can get the output data using [`data`](#tensorsdata) and [`info`](#tensorsinfo). You can use `user_data` to callback:

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+
+  ```bash
+    typedef void (*ml_pipeline_sink_cb) (const ml_tensors_data_h data, const ml_tensors_info_h info, void *user_data);
+  ```
+
+  After using it, the `sinkhandler` has to be released through `ml_pipeline_sink_unregister ()`
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase to:

Release the `sinkhandler` through `ml_pipeline_sink_unregister ()`:

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+  ```bash
+    status = ml_pipeline_sink_unregister (sinkhandle);
+  ```
+
+
+* #### Valve <a name="valveelement"></a>
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please do not use heading tags in the bullet, instead, use:

- **Valve**

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+  When the stream of a pipeline needs to be controlled, this element is useful.
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase to:
This element is used to control the stream of a pipeline:

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 7]
```diff
@@ -0,0 +1,350 @@
...
+
+  ```bash
+    gchar * pipeline = g_strdup ("videotestsrc is-live=true ! videoconvert ! videoscale ! video/x-raw,format=RGBx,width=16,height=16,framerate=10/1 ! tensor_converter ! queue ! valve name=valve1");
+    int status = ml_pipeline_construct (pipeline, NULL, NULL, &handle);
+  ```
+
+  `valve1` is the name of the valve of the pipeline, and it is opened as a default. This valve can be controlled with getting `ml_pipeline_valve_h`.
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase to:

By default, valve named `valve1` of the pipeline is opened. You can control the valve using `ml_pipeline_valve_h`:

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 8]
```diff
@@ -0,0 +1,350 @@
...
+
+  ```bash
+    ml_pipeline_h handle;
+    ml_pipeline_valve_h valve1;
+
+    status = ml_pipeline_valve_get_handle (handle, "valve1", &valve1);
+  ```
+
+  After start pipeline, the stream of a pipeline can be controlled with valve.
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase to:
After you start a pipeline, you can control the stream of the pipeline with a valve:

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 9]
```diff
@@ -0,0 +1,350 @@
...
+  ```bash
+    status = ml_pipeline_start (handle);
+
+    status = ml_pipeline_valve_set_open (valve1, true); /* close */
+  ```
+
+  It can be opened with the same way.
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase to:
You can open the pipeline by controlling the stream of a pipeline with a valve:

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 10]
```diff
@@ -0,0 +1,350 @@
...
+
+  ```bash
+    status = ml_pipeline_valve_set_open (valve1, true); /* open */
+  ```
+
+  Before destroying the pipeline, `ml_pipeline_valve_h` has to be released.
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase to:

Before you destroy the pipeline, release `ml_pipeline_valve_h`:

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 11]
```diff
@@ -0,0 +1,350 @@
...
+
+  ```bash
+    status = ml_pipeline_valve_release_handle (valve1); /* release valve handle */
+  ```
+
+* #### Switch <a name="switchelement"></a>
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please do not use heading tags in the bullet, instead, use:

- **Switch**

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 12]
```diff
@@ -0,0 +1,350 @@
...
+  When a pipeline consists of multiple branches, and only one of the branches has to be working, this element is useful.
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase to:
The switch element is used when you need only one working branch from a pipeline that has multiple branches:

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 13]
```diff
@@ -0,0 +1,350 @@
...
+  ![input-selector](./media/input-selector.png)
+
+  ```bash
+    gchar * pipeline = g_strdup ("input-selector name=ins ! tensor_converter ! tensor_sink name=sinkx "
+        "videotestsrc is-live=true ! videoconvert ! ins.sink_0 "
+        "videotestsrc num-buffers=3 is-live=true ! videoconvert ! ins.sink_1");
+
+  ```
+
+  First, getting `ml_pipeline_switch_h` is required. The name of switch in this pipeline is `ins`
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase to:

Get `ml_pipeline_switch_h`.  The name of the switch in this pipeline is `ins`:

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 14]
```diff
@@ -0,0 +1,350 @@
...
+
+  ```bash
+    ml_pipeline_h handle;
+    ml_pipeline_switch_h switchhandle;
+
+    status = ml_pipeline_switch_get_handle (handle, "ins", &type, &switchhandle);
+  ```
+
+  Now, the switch is controllable with getting handle `ml_pipeline_switch_h`.
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase to:

You can control the switch using the handle `ml_pipeline_switch_h`:

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 15]
```diff
@@ -0,0 +1,350 @@
...
+  ```bash
+    ml_pipeline_h handle;
+    ml_pipeline_switch_h switchhandle;
+    ml_pipeline_switch_e type;
+
+    status = ml_pipeline_construct (pipeline, NULL, NULL, &handle);
+    status = ml_pipeline_switch_get_handle (handle, "ins", &type, &switchhandle);
+  ```
+
+  Before destroy the pipeline, `ml_pipeline_switch_h` has to be released.
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase to:

Before you destroy the pipeline, release `ml_pipeline_switch_h`:

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 16]
```diff
@@ -0,0 +1,350 @@
...
+
+  ```bash
+    status = ml_pipeline_switch_release_handle (switchhandle);
+  ```
+
+  Switch can be located at the end of the pipeline like below image.
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase it to:
The following image shows the switch at the end of the pipeline:

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 17]
```diff
@@ -0,0 +1,350 @@
...
+  ![output-selector](./media/output-selector.png)
+
+  ```bash
+    gchar * pipeline = g_strdup ("videotestsrc is-live=true ! videoconvert ! tensor_converter ! output-selector name=outs "
+        "outs.src_0 ! tensor_sink name=sink0 async=false "
+        "outs.src_1 ! tensor_sink name=sink1 async=false");
+  ```
+
+### Pipeline States <a name="pipelinestate"></a>
+  You can get the information about the states at [Gstreamer guide](https://gstreamer.freedesktop.org/documentation/plugin-development/basics/states.html).
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase to:

For more information on the pipeline states, see [GStreamer guide](https://gstreamer.freedesktop.org/documentation/plugin-development/basics/states.html).

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 18]
```diff
@@ -0,0 +1,350 @@
...
+
+## Tensors Info <a name="tensorsinfo"></a>
+
+This is a structure which contains the information of the Tensors. With below apis, `tensor info` can be managed.
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase to:
Tensors info is a structure that contains the information of the Tensors. The `tensor info` can be managed using the following APIs:

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 19]
```diff
@@ -0,0 +1,350 @@
...
+    /* Get the name of the tensor_0 */
+    status = ml_tensors_info_get_tensor_name (info, 0, &out_name);
+
+    /* Getaks  the size of the tensor_0 */
+    status = ml_tensors_info_get_tensor_size (info, 0, &data_size);
+  ```
+
+## Tensors Data <a name="tensorsdata"></a>
+
+This is a structure which contains the raw data of tensors. With below apis, `tensor data` can be managed.
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: Please rephrase to:
Tensors data is a structure that contains the raw data of tensors. The `tensor data` can be managed using the following APIs:

File: docs/application/native/guides/machine-learning/overview.md [hunk 20]
```diff
@@ -0,0 +1,14 @@
+# Machine-Learining
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/overview.md]: # Machine Learning

File: docs/application/native/guides/machine-learning/overview.md [hunk 21]
```diff
@@ -0,0 +1,14 @@
...
+
+
+Machine-learning features introduce how you can handle machine learning frameworks, like tensorflow, tensorflow-lite, with the construction of data pipeline.
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/overview.md]: Please rephrase to:
Machine-learning features help you to handle machine learning frameworks, like TensorFlow, TensorFlow-lite, with the construction of data pipeline.

File: docs/application/native/guides/machine-learning/overview.md [hunk 22]
```diff
@@ -0,0 +1,14 @@
...
+
+You can use the following machine-learning features in your native applications:
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/overview.md]: Please make this small change:

You can use the following machine learning features in your native applications:

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 23]
```diff
@@ -0,0 +1,350 @@
...
+  ```
+
+- **Feature**
+  ```bash
+  <feature name="http://tizen.org/feature/machine_learning">true</feature>
+  <feature name="http://tizen.org/feature/machine_learning.inference">true</feature>
+  ```
+
+
+## Single API <a name="singleapi"></a>
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: As a best practice, it would be good to give a small introduction line before the steps begin.

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 24]
```diff
@@ -0,0 +1,350 @@
...
+      ml_tensors_data_create (in_info, &input);
+      ml_single_invoke (single, input, &output);
+    ```
+
+4. close the opened handle
+    ```bash
+    ml_single_close (single);
+    ```
+
+## Pipeline API <a name="pipelineapi"></a>
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: As a best practice, it would be good to give a small introduction line before the steps begin.

File: docs/application/native/guides/machine-learning/overview.md [hunk 25]
```diff
@@ -0,0 +1,14 @@
...
+
+- [Machine Learning Inference C-API](machine-learning-inference.md)
+
+  You can construct the data pipeline which contains the machine learning model filter. The pipeline can be assembled and controlled with various elements of [gstreamer](https://gstreamer.freedesktop.org/).
```
Comment (divyk) [inline docs/application/native/guides/machine-learning/overview.md]: Please make this small change, to maintain parallelism with the related document:

You can easily construct and efficiently execute data stream pipelines, which contain the machine learning model filter. The pipeline can be assembled and controlled with various elements of [GStreamer](https://gstreamer.freedesktop.org/).

File: docs/application/native/guides/machine-learning/overview.md [hunk 26]
```diff
@@ -0,0 +1,14 @@
...
+
+You can use the following machine-learning features in your native applications:
+
+- [Machine Learning Inference C-API](machine-learning-inference.md)
+
+  You can construct the data pipeline which contains the machine learning model filter. The pipeline can be assembled and controlled with various elements of [gstreamer](https://gstreamer.freedesktop.org/). 
+
+## Related Information
+- Dependencies
+  - Tizen 5.5 and Higher
```
Comment (mijin85cho) [inline docs/application/native/guides/machine-learning/overview.md]: As this page is Native guide, please follow the form used for Native guides.

  - Tizen 5.5 and Higher for Mobile
  - Tizen 5.5 and Higher for Wearable

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 27]
```diff
@@ -0,0 +1,350 @@
...
+    status = ml_tensors_data_get_tensor_data (data, 0, &data_ptr, &data_size);
+
+    /* Set tensor data */
+    uint8_t dummy[4] = {1, 1, 1, 1};
+    status = ml_tensors_data_set_tensor_data (data, 0, dummy, 1);
+  ```
+
+## Related Information
+- Dependencies
+  - Tizen 5.5 and Higher
```
Comment (mijin85cho) [inline docs/application/native/guides/machine-learning/machine-learning-inference.md]: As this page is Native guide, please follow the form used for Native guides.
Instead of "- Tizen 5.5 and Higher",

  - Tizen 5.5 and Higher for Mobile
  - Tizen 5.5 and Higher for Wearable

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 28]
```diff
@@ -0,0 +1,350 @@
+# Machine Learning Inference C-API
+
+Developers can easily construct and efficiently execute data stream pipelines consisting of neural networks as filters in pipelines.
+
+The main features of Machine Learning Inference C-API include:
```
[the remark this answers is not in the document]
Comment (mijin85cho) [reply docs/application/native/guides/machine-learning/machine-learning-inference.md]: Global : 
@divyk As this page is under Native guide, I think "C-API" is not needed.

For this sentence, for example, "The main features of the Machine Learning Inference API include:"

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 29]
```diff
@@ -0,0 +1,350 @@
...
+
+- **Construction of data pipeline based on [gstreamer](https://gstreamer.freedesktop.org/)**
```
[the remark this answers is not in the document]
Comment (mijin85cho) [reply docs/application/native/guides/machine-learning/machine-learning-inference.md]: @divyk Also, how about using default font for this list items, not bold font, as like other native guide pages?

File: docs/application/native/guides/machine-learning/machine-learning-inference.md (hunk 28 above)
[the remark this answers is not in the document]
Comment (divyk) [reply docs/application/native/guides/machine-learning/machine-learning-inference.md]: @mijin85cho , I agree with you.

File: docs/application/native/guides/machine-learning/machine-learning-inference.md (hunk 29 above)
[the remark this answers is not in the document]
Comment (divyk) [reply docs/application/native/guides/machine-learning/machine-learning-inference.md]: @mijin85cho , I agree with you.

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 30]
```diff
@@ -0,0 +1,350 @@
...
+- **Support various neural net frameworks (NNFW)**
+  
+  TensorFlow, TensorFlow-Lite, Caffe2, and PyTorch are support. Thus, neural network model files trained by such frameworks can be imported as filters of pipelines directly.
+  Custom filters, which are neural network models implemented directly with programming languages including C/C++ and Python, may be imported as filters of pipelines directly as well.
+
+  Note that, in Tizen release, this is limited to TensorFlow-Lite only; we do not ship other frameworks for official embedded device OS images.
+
+## Prerequisite <a name="prerequisite"></a>
+
+To use the APIs, below headers and features are required.
```
[the remark this answers is not in the document]
Comment (mijin85cho) [reply docs/application/native/guides/machine-learning/machine-learning-inference.md]: From line 26 to line 37, please follow: 
To enable your application to use the machine learning functionality:

1. To use the functions and data types of the Machine Learning Inference API (in \[mobile](../../api/mobile/latest/ API Link) and \[wearable](../../api/wearable/latest/ API Link) applications), include the `<nnstreamer.h>` header file in your application:

   ```
   #include <nnstreamer.h>
   ```


2. To use the Machine Learning Inference API, include the following feature in your tizen-manifest.xml file.

   ```
   <feature name="http://tizen.org/feature/machine_learning">true</feature>
   <feature name="http://tizen.org/feature/machine_learning.inference">true</feature>
   ```

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 31]
```diff
@@ -0,0 +1,350 @@
...
+      <nnstreamer-single.h>
+      
+      ml_single_h single;
+      ml_tensors_info_h in_info, out_info;
+      ...
+
+      ml_single_open (&single, "model_file_path", in_info, out_info, ML_NNFW_TYPE_TENSORFLOW_LITE, ML_NNFW_HW_ANY);
+    ```
+    
+    To load a model file, 2 `ml_tensors_info_h` are required. `in_info` contains the information of input tensor, and `out_info` is for output tensor. For the detail, you can check [Tensors Info section](#tensorsinfo).
```
[the remark this answers is not in the document]
Comment (mijin85cho) [reply docs/application/native/guides/machine-learning/machine-learning-inference.md]: @divyk 
For "see [Tensors Info section](#tensorsinfo)" 
How about just "see [Tensors Information](#tensorsinfo)"?

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 32]
```diff
@@ -0,0 +1,350 @@
...
+
+2. APIs about [Tensor info](#tensorsinfo).
```
[the remark this answers is not in the document]
Comment (mijin85cho) [reply docs/application/native/guides/machine-learning/machine-learning-inference.md]: @divyk Thanks for the modification trying.

Global.
How about using "information" instead of "info"?

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 33]
```diff
@@ -0,0 +1,350 @@
...
+
+3. Stop the pipeline
+    ```bash
+      status = ml_pipeline_stop (handle);
+      g_usleep (50000); /* 50ms. Let a few frames flow. */
+
+      status = ml_pipeline_get_state (handle, &state);
+    ```
+
+4. Pipeline destroy
```
[the remark this answers is not in the document]
Comment (mijin85cho) [reply docs/application/native/guides/machine-learning/machine-learning-inference.md]: @divyk As this line has another sentence below (line 104), I think we don't need to add : at here.

Also please check the preview. In md file one enter doesn't separate the lines. We need one more enter.
In preview this line looks like below.
image

File: docs/application/native/guides/machine-learning/machine-learning-inference.md (hunk 9 above)
Comment (mijin85cho) [reply docs/application/native/guides/machine-learning/machine-learning-inference.md]: @divyk How about 
"You can also open the pipeline by controlling the stream of a pipeline with a valve:"?

File: docs/application/native/guides/machine-learning/machine-learning-inference.md (hunk 31 above)
[the remark this answers is not in the document]
Comment (divyk) [reply docs/application/native/guides/machine-learning/machine-learning-inference.md]: @mijin85cho , I agree.

File: docs/application/native/guides/machine-learning/machine-learning-inference.md (hunk 33 above)
[the remark this answers is not in the document]
Comment (divyk) [reply docs/application/native/guides/machine-learning/machine-learning-inference.md]: @mijin85cho , I agree.

File: docs/application/native/guides/machine-learning/machine-learning-inference.md (hunk 9 above)
Comment (divyk) [reply docs/application/native/guides/machine-learning/machine-learning-inference.md]: @mijin85cho , I agree.

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 34]
```diff
@@ -0,0 +1,350 @@
...
+
+  You can compose the data stream pipeline through Machine Learning Inference C-API with various elements of gstreamer and nnstreamer.
+
+- **[Single](#singleapi) & [Pipeline](#pipelineapi) API**
+  
+  There are 2 types of Machine Learning Inference C-API. Single API is useful for simple usage scenario of neural network models. It allows to invoke a neural network model with a single instance of input data for the model directly. It is useful if you have the input data pre-processed with the application itself and there is no complex interactions between neural network models, data processors, or data stream paths.
+  
+  On the other hand, Pipeline API allows developers to construct and execute pipelines with multiple neural network models, multiple input and output nodes, multiple data processors, pre-and-post processors, and various data path manipulators. Besides, if the input is online data (or streamed data), Pipeline APIs simplify your application and improve its performance.
+  
+- **Support various neural net frameworks (NNFW)**
```
[the remark this answers is not in the document]
Comment (divyk) [reply docs/application/native/guides/machine-learning/machine-learning-inference.md]: @mijin85cho , is Supportability fine?

Supportability of various neural net frameworks (NNFW)

File: docs/application/native/guides/machine-learning/machine-learning-inference.md [hunk 35]
```diff
@@ -0,0 +1,350 @@
...
+  
+  TensorFlow, TensorFlow-Lite, Caffe2, and PyTorch are support. Thus, neural network model files trained by such frameworks can be imported as filters of pipelines directly.
+  Custom filters, which are neural network models implemented directly with programming languages including C/C++ and Python, may be imported as filters of pipelines directly as well.
+
+  Note that, in Tizen release, this is limited to TensorFlow-Lite only; we do not ship other frameworks for official embedded device OS images.
```
[the remark this answers is not in the document]
Comment (journey2w) [reply docs/application/native/guides/machine-learning/machine-learning-inference.md]: @divyk Tizen is open source.  Samsung must not be mentioned in any Tizen developer's guides.

> **Note**
>
> The devices powered by Tizen OS can contain TensorFlow-Lite only. Ensure that the neural network frameworks that you want to use are installed.

[the remark this answers is not in the document]
Comment (divyk) [reply docs/application/native/guides/machine-learning/machine-learning-inference.md]: @journey2w , thanks for the correction. I understand, will keep in mind for future documents. Can you please suggest what can replace the third-person usage here?

File: docs/application/native/guides/machine-learning/machine-learning-inference.md (hunk 34 above)
[the remark this answers is not in the document]
Comment (mijin85cho) [reply docs/application/native/guides/machine-learning/machine-learning-inference.md]: @divyk I think below would be enough

Support for various neural net frameworks (NNFW)

**Comment by helloahn** [comment]
I reviewed the [commit](https://github.com/Samsung/tizen-docs/pull/893/commits/ac26e01bf5c6baa9df2440f374758cbdee7527ec) by @journey2w. Thanks for your help.
